bfasst/ninja_flows/vivado.py

```diff
@@ -0,0 +1,32 @@
+"""Flow to create Vivado synthesis and implementation ninja snippets."""
+from bfasst.ninja_flows.flow import Flow
+from bfasst.paths import NINJA_FLOWS_PATH
+
+
+class Vivado(Flow):
+    """Flow to create Vivado synthesis and implementation ninja snippets."""
+
+    def __init__(self, design, flow_args=None, ooc=False):
+        super().__init__()
+        self.ooc = ooc
+        self.vivado_tool = self.configure_vivado_tool(design, flow_args, ooc)
+
+    def create_rule_snippets(self):
+        self.vivado_tool.create_rule_snippets()
+
+    def create_build_snippets(self):
+        self.vivado_tool.create_build_snippets()
+
+    def add_ninja_deps(self, deps=None):
+        if not deps:
+            deps = []
+        deps.extend(self.vivado_tool.add_ninja_deps())
+        deps.append(f"{NINJA_FLOWS_PATH}/vivado.py ")
+        if self.ooc:
+            deps.append(f"{NINJA_FLOWS_PATH}/vivado_ooc.py ")
```

Comment:
Are deps always paths?  If so, we should use `Path` not `str`

Reply:
They are paths, but they are templated into the build.ninja file, and so need to be strings, not paths. These deps are used to determine when to rebuild the build.ninja file. If any of them change, the build.ninja file will rerender itself. But we have to template them into the file as strings.